pkg/controller/handlers/slackreceiver/slackreceiver.go

```diff
@@ -47,3 +71,149 @@ func CreateOAuthApp(req router.Request, _ router.Response) error {
 
 	return nil
 }
+
+func (s *Handler) SubscribeToSlackEvents(req router.Request, resp router.Response) error {
+	slackReceiver := req.Object.(*v1.SlackReceiver)
+
+	cred, err := s.gptScript.RevealCredential(req.Ctx, []string{system.OAuthAppPrefix + slackReceiver.Spec.ThreadName}, string(types.OAuthAppTypeSlack))
+	if err != nil {
+		return err
+	}
+
+	slackAppToken := cred.Env["APP_TOKEN"]
+
+	if slackAppToken == "" {
+		return nil
+	}
+
+	if s.subscribed[slackReceiver.Name] != nil {
+		return nil
+	}
```

Comment:
I recommend moving this up to the top of the function before we try to reveal the credential, as it could save us from some unnecessary `Reveal` calls.